File: Breadth-First Search/Binary tree/LET_no429_N-ary Tree Level Order Traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if not root:
            return []
        
        queue = collections.deque([root])
        res = []
        
        while queue:
            cur_level = []
            for _ in range(len(queue)):
                node = queue.popleft()
                cur_level.append(node.val)
                for child in node.children:
                    queue.append(child)
            res.append(cur_level[:])
        
        return res

File: Breadth-First Search/Binary tree/LET_no429. N-ary Tree Level Order Traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        if (!root)
            return {};
        
        vector<vector<int>> res;
        queue<Node*> q;
        q.push(root);
            
        while (!q.empty()) {
            vector<int> cur_level;
            
            int n = q.size();
            for (int i = 0; i < n; i++) {
                Node* cur = q.front();
                q.pop();
                cur_level.push_back(cur->val);
                for (Node* child : cur->children)
                    q.push(child);
            }
            
            res.push_back(cur_level);
        }
        
        return res;
    }
};